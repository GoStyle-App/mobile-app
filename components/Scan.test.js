import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { render } from '@testing-library/react-native';
import Scan from './Scan';

describe('Testing Scan component', () => {
    const navigation = {
        'addListener': function addListener(){},
        'canGoBack': function canGoBack(){},
        'closeDrawer': function anonymous(){},
        'dangerouslyGetParent': function dangerouslyGetParent(){},
        'dangerouslyGetState': function anonymous(){},
        'dispatch': function dispatch(){},
        'goBack': function anonymous(){},
        'isFocused': function isFocused(){},
        'jumpTo': function anonymous(){},
        'navigate': function anonymous(){},
        'openDrawer': function anonymous(){},
        'removeListener': function removeListener(){},
        'reset': function anonymous(){},
        'setOptions': function setOptions(){},
        'setParams': function anonymous(){},
        'toggleDrawer': function anonymous(){},
    };

    test('Correct render of component', () => {
        const component = (
            <NavigationContainer>
                <Scan navigation={ navigation } />
            </NavigationContainer>
        );
        const ScanRender = render(component);

        expect(ScanRender).toMatchSnapshot();
    });
});
// TODO : faire des tests pour Scan
