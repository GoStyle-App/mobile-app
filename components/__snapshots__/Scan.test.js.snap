// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Scan component Correct render of component 1`] = `
<Text>
  Requesting for camera permission
</Text>
`;
